lua_modules/sntp/sntp.lua

```diff
@@ -0,0 +1,226 @@
+local MAX_SERVER_ATTEMPTS = 2
+local SNTP_TIMEOUT        = 5000
+
+local defserv = {
```

Comment:
There is a good argument for making this JiT to avoid having the array permanently allocated and bound to the upval `defserv`.  E.g. 
```Lua
local function defserv()
 return {"0.nodemcu.pool.ntp.org",  "1.nodemcu.pool.ntp.org",  "2.nodemcu.pool.ntp.org", "3.nodemcu.pool.ntp.org"}
end
```
and using `serv = defserv()` below

Reply:
Sorry, can you explain that to me in more detail?  In this case, why is it better to point at a thunk that will build the table than at the table itself?

Reply:
What the original for does is to build a 4-entry table during the require evaluation and store it in the local `defserv` which is kept as a closed upval in the constructor.  The function version keeps a closure `defserv`  as a closed upval in the constructor. The table is built JiT. In practice, I suspect that most uses will use this default, so this isn't really a saving.  In terms of saving RAM, the best approach would be to move the default array inline.

Reply:
Oh, I see.  OK, inline it shall go.